phantazm-mob/src/main/java/com/github/phantazmnetwork/mob/target/SerializableMappedSelector.java

```diff
@@ -0,0 +1,21 @@
+package com.github.phantazmnetwork.mob.target;
+
+import com.github.phantazmnetwork.api.config.VariantSerializable;
+import net.kyori.adventure.key.Key;
+import org.jetbrains.annotations.NotNull;
+
+public abstract class SerializableMappedSelector<TFrom, TTo> extends MappedSelector<TFrom, TTo> implements VariantSerializable {
+
+
+    public SerializableMappedSelector(@NotNull TargetSelector<TFrom> delegate) {
+        super(delegate);
+        if (!(delegate instanceof VariantSerializable)) {
+            throw new IllegalArgumentException("delegate is not variant serializable");
+        }
+    }
+
+    @Override
+    public @NotNull Key getSerialKey() {
```

Comment:
any reason why this functionality can't be moved up to MappedSelector (making all MappedSelectors VariantSerializable) and removing the need for this class? 
or is there a specific reason you might want *some* classes to be non-variant-serializable

Reply:
in general I don't think base target selectors *need* to be serializable for mock code testing purposes, but we could enforce it

Reply:
can't mock code just return a useless Key, e.g. Key.key("testing", "mock_selector")

Reply:
it can, wouldn't be the end of the world and would get rid of some casting if it's a lesser evil